screens/Home/HomeScreen.tsx

```diff
@@ -11,153 +11,98 @@ import { useSafeAreaInsets } from "react-native-safe-area-context"
-    latitude: 46.51858962578904,
-    longitude: 6.566048509782951,
-    description: "Music festival 2023",
-    date: "2023-04-04", // changed to YYYY-MM-DD format
-    imageUrl: "https://balelec.ch/uploads/2023_f4caf642b6.jpg",
-  },
-  {
-    title: "Event 2",
-    location: "EPFL, CM",
-    latitude: 46.51859297638995, 
-    longitude: 6.563141941352757,
-    description: "Agepoly",
-    date: "2022-08-04", // changed to YYYY-MM-DD format
-    imageUrl: "https://example.com/image.png",
-  },
-  {
-    title: "Balelek 2024",
-    location: "EPFL, Agora",
-    latitude: 46.51992048733761,
-    longitude: 6.565926796074852,
-    description: "Music festival 2024",
-    date: "2024-05-20", // changed to YYYY-MM-DD format
-    imageUrl: "https://example.com/image.png",
-  },
-  {
-    title: "Graduation 2025",
-    location: "EPFL, Swisstech",
-    latitude: 46.52277976263842,
-    longitude: 6.565209257647138,
-    description: "Graduation",
-    date: "2025-06-04", // changed to YYYY-MM-DD format
-    imageUrl: "https://example.com/image.png",
-  },
-]
 
 const HomeScreen = () => {
   const insets = useSafeAreaInsets()
   const navigation = useNavigation()
-  //const [title, setTitle] = React.useState("Future")
-
-  const [searchQuery, setSearchQuery] = React.useState("")
-  // Filter events based on the current date and search query
-  const getFilteredEvents = (isFutureEvent: boolean) => {
-    const currentDate = new Date("2024-04-20")
+  const [futureEvents, setFutureEvents] = React.useState<Event[] | null>([])
+  const [pastEvents, setPastEvents] = React.useState<Event[] | null>([])
 
-    // Filter events based on date
-    const eventsFilteredByDate = events.filter((event) => {
-      const eventDate = new Date(event.date)
-      return isFutureEvent ? eventDate >= currentDate : eventDate < currentDate
-    })
+  const [ setSearchQuery] = React.useState("")
 
-    // Further filter by search query if there is one
-    return eventsFilteredByDate.filter((event) =>
-      event.title.toLowerCase().includes(searchQuery.toLowerCase())
-    )
+  const handleSearch = (search: string) => {
+    setSearchQuery(search)
   }
+  useEffect(() => {
+    const loadEvents = async () => {
+      try {
+        const fetchedFutureEvents = await getAllFutureEvents() 
+        const fetchedPastEvents = await getAllPastEvents() 
 
-  const futureEvents = getFilteredEvents(true)
-  const pastEvents = getFilteredEvents(false)
+        setFutureEvents(fetchedFutureEvents)
+        setPastEvents(fetchedPastEvents)
+      }
+      catch (error) {
+        showErrorToast("Error fetching events. Please check your connection and try again.")
+      }
+    }
 
-  const handleSearch = (search: string) => {
-    setSearchQuery(search)
+    loadEvents()
+  }, [])
+
+  const renderEvents = (isFuture: boolean) => {
+    const events: Event[] = isFuture ? futureEvents : pastEvents
+    return events.map((event: Event) => (
+      <TouchableOpacity key={event.title}>
```

Comment:
Here I think you should use the event uid as the key because it can be a problem when two events have the same name and thus the same key.